pkg/daemon/daemon.go

```diff
@@ -200,10 +200,14 @@ func launchToken(
 	cmds := strings.Split(text, "||")
 
 	for i, cmd := range cmds {
-		err := launcher.LaunchToken(cfg, cfg.GetAllowCommands() || mapped, kbd, cmd, len(cmds), i)
+		err, softwareSwap := launcher.LaunchToken(cfg, cfg.GetAllowCommands() || mapped, kbd, cmd, len(cmds), i)
 		if err != nil {
 			return err
 		}
+		if softwareSwap {
+			log.Info().Msgf("current software launched set to: %s", token.UID)
+			state.SetCurrentlyLoadedSoftware(token.UID)
+		}
```

Comment:
LaunchToken will return an extra boolean that we use to track if the current scanned token has changed the software running. If yes we keep the UID memorized.
The token.UID is used to both know that a software didn't change and that we are putting back a card with the same software. is used to AVOID triggering exit_game on card removals and swap.